### ISSUE TITLE
program hung when http response status-code is 404 not found

### ISSUE THREAD

**Issue by heartszhang** [vbauerster/getparty#2]
```sh
getparty http://xxx.xxx.xx.xx/gfssrc/media/linda/20180331/v20746341/v20746341.mp4
```
>  [ 2018/04/13 17:41:41 ] http://xxx.xxx.xx.xx/gfssrc/media/linda/20180331/v20746341/v20746341.mp4
> HTTP request sent, awaiting response... 404 Not Found

`kill -9` is the only way to terminate the program

**Comment by vbauerster** [comment]
Thanks for reporting! I've made big refactoring, so this issue should be fixed now. Can you please check?

**Comment by vbauerster** [comment]
Fixed in [v1.3.0](https://github.com/vbauerster/getparty/releases/tag/v1.3.0) release. Feel free to open, if issue reoccurs.
